.github/workflows/ci.yml

```diff
@@ -33,26 +33,27 @@ jobs:
           elixir-version: ${{matrix.pair.elixir}}
 
       - uses: actions/cache@v2
+        id: cache-deps
         with:
           path: |
             deps
             _build
           key: mix-${{ runner.os }}-${{matrix.pair.elixir}}-${{matrix.pair.otp}}-${{ hashFiles('**/mix.lock') }}
 
-      - run: mix deps.get
+      - run: mix do deps.get, deps.compile
+        if: steps.cache-deps.outputs.cache-hit != 'true'
```

Comment:
We were not using the cache for anything. We are caching the compiled dependencies for the OS/Elixir/OTP/lockfile combination, so it's safe to cache the _compiled_ artifacts. In turn, this means we can skip `deps.get` and `deps.compile` altogether if the cache hits.